tests/unit/test_streamlit.py

```
@@ -44,3 +44,46 @@ def test_fetch_streamlit_apps(self):
+
+    def test_skips_app_when_directory_missing(self):
+        """Apps whose parent directory does not exist are skipped without crashing."""
+        apps = self._make_apps("deleted_folder/app.py", "existing_folder/app.py")
+        mock_venv = MagicMock()
+        mock_logger = MagicMock(spec=logging.Logger)
+
+        def exists_side_effect(path):
+            return "deleted_folder" not in path
+
+        with patch("installer.module.streamlit.fetch_streamlit_apps", return_value=apps):
+            with patch("installer.module.streamlit.os.path.exists", side_effect=exists_side_effect):
```

Ruff (0.15.17)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)